src/server/set_family.cc

```diff
@@ -1378,6 +1381,36 @@ void SInterStore(CmdArgList args, ConnectionContext* cntx) {
   (*cntx)->SendLong(result->size());
 }
 
+void SInterCard(CmdArgList args, ConnectionContext* cntx) {
+  unsigned num_keys;
+  if (!absl::SimpleAtoi(ArgS(args, 0), &num_keys)) {
+    return (*cntx)->SendError("numkeys should be greater than 0");
+  } else if (args.size() == (num_keys + 3) && num_keys > args.size() - 3) {
```

Comment:
As @dranikpg pointed out the `args.size() == (num_keys + 3) && num_keys > args.size() - 3)` is problematic because the conjunction of the expressions: `args.size() - 3 == num_keys` and  `num_keys > args.size()  - 3` can't be both true at the same time.

I would slightly rewrite this condition, basically I would think of it as:
1. Either the size of the arguments should be equal to the number of keys 
2. For the LIMIT case, the size of the arguments should be equal to the number of keys plus 2 (because of limit) 

If non of 1 or 2 holds, then print error which formally is:

`if !(cond 1 || cond 2)` then print error